dynamiq/nodes/agents/react.py

```diff
@@ -1309,13 +1309,39 @@ def _run_agent(
                     self._prompt.messages.append(Message(role=MessageRole.USER, content=observation, static=True))
 
                     if self.streaming.enabled and self.streaming.mode == StreamingMode.ALL:
+                        if "tool" in locals():
```

Comment:
Please rewrite the logic to don't use `locals`. You could define `tools` with default value before